app/main.py

```
@@ -1,3 +1,59 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float) -> None:
+        self.km: float = km
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __add__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, (int, float)):
+            return Distance(km=self.km + other)
+        return Distance(km=self.km + other.km)
+
+    def __iadd__(self, other: int | float | Distance) -> Distance:
+        if isinstance(other, Distance):
+            self.km += other.km
+        else:
+            self.km += other
+        return self
+
+    def __mul__(self, other: int | float) -> Distance:
```

The implementation of `__mul__` can return None for non-numeric `other` because the method only returns inside the `if isinstance(other, (int, float))` branch and has no final return. This violates the requirement: **"All arithmetic methods must return a Distance instance (for __add__, __mul__, __truediv__) or modify self appropriately (__iadd__)."**

Suggested fixes:
- Remove the `isinstance` guard and simply `return Distance(km=self.km * other)`, which lets Python raise a `TypeError` for unsupported operand types (this aligns with the checklist guidance that explicit runtime checks are not required when annotations are present). OR
- Keep the guard but add an explicit `raise TypeError(...)` in the non-numeric branch so the method never returns `None`.

Either approach will ensure the method always returns a `Distance` or raises an appropriate error.